tpm2/update.go

```diff
@@ -43,13 +42,33 @@ var skdbUpdatePCRProtectionPolicyImpl = (*sealedKeyDataBase).updatePCRProtection
 //
 // If k.data.policy().pcrPolicyCounterHandle() is not tpm2.HandleNull, then counterPub
 // must be supplied, and it must correspond to the public area associated with that handle.
-func (k *sealedKeyDataBase) updatePCRProtectionPolicyImpl(tpm *tpm2.TPMContext, key secboot.PrimaryKey,
-	counterPub *tpm2.NVPublic, profile *PCRProtectionProfile, session tpm2.SessionContext) error {
+func (k *sealedKeyDataBase) updatePCRProtectionPolicyNoValidate(tpm *tpm2.TPMContext, key secboot.PrimaryKey,
+	counterPub *tpm2.NVPublic, profile *PCRProtectionProfile, incrementPolicyVersion bool,
```

Comment:
using a bool for incrementPolicyVersion is not very readable, we usually work around that keeping the bool type but always passing a const in the invocations:

const incrementPolicyVersion = true|false
updatePCRProtectionPolicyNoValidate(..., incrementPolicyVersion, ...)

or we need to switch the param to an enum or pass an option struct, the latter is not a obvious change here though

Reply:
I've cleaned this up a bit with an enumerated type. I also ended up restoring the old behaviour for the `SealedKeyObject` type where every update increments the counter as opposed to changing the behaviour in this PR, so this function has 3 options now.